client.go

```diff
@@ -159,6 +161,18 @@ func (c *Client) Authenticate(profile, scope string) (*AuthInfo, string, error)
 	}, authURL, nil
 }
 
+// newState generates a new state value.
+func newState() string {
+	// OAuth 2.0 requires state to be printable ASCII, so base64 fits.
+	// See https://datatracker.ietf.org/doc/html/rfc6749#appendix-A.5.
+	b := make([]byte, 64)
+	_, err := cryptorand.Read(b)
+	if err != nil {
+		panic(err)
```

Comment:
I don't feel particularly safe panicking here. I understand that this should never happen, but why panic and not return an error?

Reply:
Hmm, my rationale to handle the error this way has been that I couldn't think of a scenario where the source of entry on the system would be exhausted to a point of getting an error from `crypto/rand.Read`, and there being anything useful to do other than abruptly stopping execution due to an internal error. (I've not observed any occurrences so far, and so never had the need to revisit this simple error handling strategy.)

Returning an error to the user is another viable strategy, though it incurs the cost onto the user of needing to think about it. Since this is a library, not a command, and it's possible some users might want to have finer control over how this error is handled, that might be a good path to take.

Changed it to return the error to the caller, PTAL.